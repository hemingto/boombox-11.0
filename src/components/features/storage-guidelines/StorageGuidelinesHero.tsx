/**
 * @fileoverview Storage guidelines page hero section component
 * @source boombox-10.0/src/app/components/storage-guidelines/storageguidelineshero.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays the hero section for the storage guidelines page with heading and subtitle.
 * Uses semantic HTML and responsive padding following design system standards.
 * 
 * API ROUTES UPDATED:
 * - None (presentational component only)
 * 
 * DESIGN SYSTEM UPDATES:
 * - Uses existing responsive padding patterns (lg:px-16 px-6)
 * - Maintains semantic spacing with my-10 sm:my-20
 * - Semantic h1 and p elements for proper content hierarchy
 * 
 * @refactor Renamed file to PascalCase, added comprehensive documentation
 */

'use client';

import React from 'react';

/**
 * StorageGuidelinesHero component
 * 
 * Simple hero section for the storage guidelines page displaying the page title
 * and a subtitle describing the content (best practices).
 * Uses responsive padding and spacing following the design system.
 * 
 * @example
 * ```tsx
 * <StorageGuidelinesHero />
 * ```
 */
export function StorageGuidelinesHero(): React.ReactElement {
  return (
    <div className="w-full my-10 sm:my-20 lg:px-16 px-6">
      <div>
        <h1 className="mb-4">Storage Guidelines</h1>
        <p>best practices</p>
      </div>
    </div>
  );
}

