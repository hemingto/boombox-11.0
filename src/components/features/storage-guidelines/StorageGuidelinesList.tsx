/**
 * @fileoverview Storage guidelines list component displaying numbered guidelines with descriptions
 * @source boombox-10.0/src/app/components/storage-guidelines/storageguidelineslist.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays all storage guidelines as numbered cards with titles and detailed descriptions.
 * Each guideline includes rules, best practices, and safety information for using Boombox storage.
 * Data extracted to @/data/storageGuidelines for better maintainability.
 * 
 * API ROUTES UPDATED:
 * - None (presentational component only)
 * 
 * DESIGN SYSTEM UPDATES:
 * - border-slate-100 → border-border (semantic border color)
 * - border-zinc-950 → border-primary (semantic primary color for numbered badges)
 * - text-zinc-950 → text-primary (semantic text color for badges)
 * - Maintains responsive padding (lg:px-16 px-6)
 * - Preserves spacing and layout patterns
 * 
 * @refactor Extracted data to centralized data folder, renamed to PascalCase, applied design system colors,
 * removed hardcoded Help Center link (will be handled by parent component)
 */

import React from 'react';
import Link from 'next/link';
import { storageGuidelines } from '@/data/storageGuidelines';
import type { StorageGuideline } from '@/data/storageGuidelines';

/**
 * StorageGuidelinesList component
 * 
 * Displays a list of storage guidelines as numbered cards.
 * Each guideline includes a number badge, title, and detailed description.
 * The descriptions may contain multiple paragraphs and inline links.
 * 
 * @example
 * ```tsx
 * <StorageGuidelinesList />
 * ```
 */
export function StorageGuidelinesList(): React.ReactElement {
  return (
    <div className="lg:px-16 px-6">
      {storageGuidelines.map((guideline: StorageGuideline) => (
        <article 
          key={guideline.number} 
          className="flex mb-8 p-6 border border-border rounded-md items-start"
          aria-labelledby={`guideline-${guideline.number}-title`}
        >
          {/* Number Badge */}
          <div 
            className="flex mr-4 mt-0.5 items-center justify-center w-8 h-8 rounded-full flex-shrink-0 border-2 border-primary text-primary font-semibold"
            aria-label={`Guideline ${guideline.number}`}
          >
            {guideline.number}
          </div>
          
          {/* Content */}
          <div>
            <h2 id={`guideline-${guideline.number}-title`} className="mb-5">
              {guideline.title}
            </h2>
            <div className="text-text-primary">
              {/* Special handling for guideline 2 to add Help Center link */}
              {guideline.number === 2 ? (
                <>
                  <p className="mb-2">All items need to fit into our standard 5x8 Boombox storage container. The door to your unit must be able to close properly. If the door is unable to be properly closed you must remove the stored item that is preventing the door from closing.</p>
                  <p>If you are unsure if one of your storage items will fit please reach out at our <Link href="/help-center" className="font-bold underline">Help Center</Link>.</p>
                </>
              ) : (
                guideline.description
              )}
            </div>
          </div>
        </article>
      ))}
    </div>
  );
}

