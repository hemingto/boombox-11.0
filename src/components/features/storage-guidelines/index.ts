/**
 * @fileoverview Storage guidelines feature components exports
 * @source boombox-10.0/src/app/components/storage-guidelines/
 */

export { StorageGuidelinesHero } from './StorageGuidelinesHero';
export { StorageGuidelinesList } from './StorageGuidelinesList';

