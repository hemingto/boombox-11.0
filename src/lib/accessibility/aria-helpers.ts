/**
 * @fileoverview ARIA helper utilities for accessibility
 * Consistent patterns for ARIA attributes, focus management, and screen reader support
 */

/**
 * Generate unique IDs for ARIA relationships
 */
export function generateAriaId(prefix: string = 'aria'): string {
  return `${prefix}-${Math.random().toString(36).substr(2, 9)}`;
}

/**
 * ARIA label helpers for common patterns
 */
export const ariaLabels = {
  /**
   * Loading state labels
   */
  loading: {
    'aria-live': 'polite' as const,
    'aria-busy': true,
    'aria-label': 'Loading content',
  },
  
  /**
   * Error state labels
   */
  error: {
    'aria-live': 'assertive' as const,
    role: 'alert' as const,
  },
  
  /**
   * Success state labels
   */
  success: {
    'aria-live': 'polite' as const,
    role: 'status' as const,
  },
  
  /**
   * Form field labels
   */
  required: {
    'aria-required': true,
  },
  
  invalid: {
    'aria-invalid': true,
  },
  
  /**
   * Navigation labels
   */
  skipLink: {
    'aria-label': 'Skip to main content',
  },
  
  breadcrumb: {
    'aria-label': 'Breadcrumb navigation',
  },
  
  /**
   * Modal/Dialog labels
   */
  modal: {
    role: 'dialog' as const,
    'aria-modal': true,
  },
  
  modalOverlay: {
    'aria-hidden': true,
  },
  
  /**
   * Button labels
   */
  closeButton: {
    'aria-label': 'Close',
  },
  
  menuButton: {
    'aria-haspopup': true,
    'aria-expanded': false,
  },
} as const;

/**
 * Focus management utilities
 */
export const focusHelpers = {
  /**
   * Trap focus within an element
   */
  trapFocus(container: HTMLElement) {
    const focusableElements = container.querySelectorAll(
      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
    );
    
    const firstElement = focusableElements[0] as HTMLElement;
    const lastElement = focusableElements[focusableElements.length - 1] as HTMLElement;
    
    const handleTabKey = (e: KeyboardEvent) => {
      if (e.key !== 'Tab') return;
      
      if (e.shiftKey) {
        if (document.activeElement === firstElement) {
          lastElement.focus();
          e.preventDefault();
        }
      } else {
        if (document.activeElement === lastElement) {
          firstElement.focus();
          e.preventDefault();
        }
      }
    };
    
    container.addEventListener('keydown', handleTabKey);
    
    // Focus first element
    firstElement?.focus();
    
    // Return cleanup function
    return () => {
      container.removeEventListener('keydown', handleTabKey);
    };
  },
  
  /**
   * Restore focus to previously focused element
   */
  restoreFocus(previousElement: HTMLElement | null) {
    if (previousElement && typeof previousElement.focus === 'function') {
      previousElement.focus();
    }
  },
  
  /**
   * Move focus to element and announce to screen readers
   */
  announceFocus(element: HTMLElement, message?: string) {
    element.focus();
    
    if (message) {
      const announcement = document.createElement('div');
      announcement.setAttribute('aria-live', 'polite');
      announcement.setAttribute('aria-atomic', 'true');
      announcement.className = 'sr-only';
      announcement.textContent = message;
      
      document.body.appendChild(announcement);
      
      setTimeout(() => {
        document.body.removeChild(announcement);
      }, 1000);
    }
  },
};

/**
 * Screen reader utilities
 */
export const screenReaderHelpers = {
  /**
   * Create screen reader only text
   */
  createSROnlyText(text: string): HTMLSpanElement {
    const span = document.createElement('span');
    span.className = 'sr-only';
    span.textContent = text;
    return span;
  },
  
  /**
   * Announce message to screen readers
   */
  announce(message: string, priority: 'polite' | 'assertive' = 'polite') {
    const announcement = document.createElement('div');
    announcement.setAttribute('aria-live', priority);
    announcement.setAttribute('aria-atomic', 'true');
    announcement.className = 'sr-only';
    announcement.textContent = message;
    
    document.body.appendChild(announcement);
    
    setTimeout(() => {
      document.body.removeChild(announcement);
    }, 1000);
  },
  
  /**
   * Update screen reader announcement
   */
  updateAnnouncement(element: HTMLElement, message: string) {
    element.textContent = message;
  },
};

/**
 * Keyboard navigation helpers
 */
export const keyboardHelpers = {
  /**
   * Handle arrow key navigation in lists/menus
   */
  handleArrowNavigation(
    event: KeyboardEvent,
    items: HTMLElement[],
    currentIndex: number,
    options: {
      loop?: boolean;
      vertical?: boolean;
      horizontal?: boolean;
    } = {}
  ) {
    const { loop = true, vertical = true, horizontal = false } = options;
    let newIndex = currentIndex;
    
    switch (event.key) {
      case 'ArrowDown':
        if (vertical) {
          newIndex = loop && currentIndex === items.length - 1 ? 0 : Math.min(currentIndex + 1, items.length - 1);
          event.preventDefault();
        }
        break;
      case 'ArrowUp':
        if (vertical) {
          newIndex = loop && currentIndex === 0 ? items.length - 1 : Math.max(currentIndex - 1, 0);
          event.preventDefault();
        }
        break;
      case 'ArrowRight':
        if (horizontal) {
          newIndex = loop && currentIndex === items.length - 1 ? 0 : Math.min(currentIndex + 1, items.length - 1);
          event.preventDefault();
        }
        break;
      case 'ArrowLeft':
        if (horizontal) {
          newIndex = loop && currentIndex === 0 ? items.length - 1 : Math.max(currentIndex - 1, 0);
          event.preventDefault();
        }
        break;
      case 'Home':
        newIndex = 0;
        event.preventDefault();
        break;
      case 'End':
        newIndex = items.length - 1;
        event.preventDefault();
        break;
    }
    
    if (newIndex !== currentIndex) {
      items[newIndex]?.focus();
      return newIndex;
    }
    
    return currentIndex;
  },
  
  /**
   * Handle escape key to close modals/dropdowns
   */
  handleEscapeKey(event: KeyboardEvent, callback: () => void) {
    if (event.key === 'Escape') {
      callback();
      event.preventDefault();
    }
  },
};

/**
 * Color contrast utilities
 */
export const contrastHelpers = {
  /**
   * Calculate color contrast ratio
   */
  getContrastRatio(color1: string, color2: string): number {
    const getLuminance = (color: string) => {
      // Convert hex to RGB
      const hex = color.replace('#', '');
      const r = parseInt(hex.substr(0, 2), 16) / 255;
      const g = parseInt(hex.substr(2, 2), 16) / 255;
      const b = parseInt(hex.substr(4, 2), 16) / 255;
      
      // Calculate relative luminance
      const sRGB = [r, g, b].map(c => {
        return c <= 0.03928 ? c / 12.92 : Math.pow((c + 0.055) / 1.055, 2.4);
      });
      
      return 0.2126 * sRGB[0] + 0.7152 * sRGB[1] + 0.0722 * sRGB[2];
    };
    
    const lum1 = getLuminance(color1);
    const lum2 = getLuminance(color2);
    const brightest = Math.max(lum1, lum2);
    const darkest = Math.min(lum1, lum2);
    
    return (brightest + 0.05) / (darkest + 0.05);
  },
  
  /**
   * Check if color combination meets WCAG standards
   */
  meetsWCAG(color1: string, color2: string, level: 'AA' | 'AAA' = 'AA'): boolean {
    const ratio = this.getContrastRatio(color1, color2);
    return level === 'AA' ? ratio >= 4.5 : ratio >= 7;
  },
}; 