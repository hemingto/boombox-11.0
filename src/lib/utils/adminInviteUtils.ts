/**
 * @fileoverview Admin invitation utility functions
 * @source boombox-10.0/src/app/api/admin/invites/route.ts (extracted inline functions)
 * @refactor Extracted admin invitation logic into reusable utility functions
 */

import { prisma } from '@/lib/database/prismaClient';
import crypto from 'crypto';
import { AdminRole } from '@prisma/client';

/**
 * Validate admin role is valid for invitation
 */
export function isValidAdminRole(role: string): role is AdminRole {
  return role === 'ADMIN' || role === 'SUPERADMIN';
}

/**
 * Generate secure random token for admin invitations
 * @returns 64-character hex token (256 bits of entropy)
 */
export function generateInviteToken(): string {
  return crypto.randomBytes(32).toString('hex');
}

/**
 * Calculate expiration date for admin invitations
 * @param daysFromNow - Number of days from current time (default: 15)
 * @returns Date object representing expiration time
 */
export function calculateInviteExpiration(daysFromNow: number = 15): Date {
  return new Date(Date.now() + daysFromNow * 24 * 60 * 60 * 1000);
}

/**
 * Check if admin already exists with given email
 * @param email - Email address to check
 * @returns Promise<boolean> - True if admin exists, false otherwise
 */
export async function checkAdminExists(email: string): Promise<boolean> {
  const existingAdmin = await prisma.admin.findUnique({
    where: { email },
  });
  return !!existingAdmin;
}

/**
 * Check if active invitation already exists for email
 * @param email - Email address to check
 * @returns Promise<boolean> - True if active invite exists, false otherwise
 */
export async function checkActiveInviteExists(email: string): Promise<boolean> {
  const existingInvite = await prisma.adminInvite.findFirst({
    where: {
      email,
      expiresAt: {
        gt: new Date(),
      },
    },
  });
  return !!existingInvite;
}

/**
 * Create admin invitation record in database
 * @param data - Invitation data
 * @returns Promise<AdminInvite> - Created invitation record
 */
export async function createAdminInvitation(data: {
  email: string;
  token: string;
  role: AdminRole;
  expiresAt: Date;
  invitedById: number;
}) {
  return await prisma.adminInvite.create({
    data,
  });
}

/**
 * Delete admin invitation by ID (used for cleanup on email failure)
 * @param inviteId - ID of invitation to delete
 * @returns Promise<void>
 */
export async function deleteAdminInvitation(inviteId: number): Promise<void> {
  await prisma.adminInvite.delete({
    where: { id: inviteId },
  });
}

/**
 * Format role for display in emails and UI
 * @param role - AdminRole enum value
 * @returns Formatted role string for display
 */
export function formatRoleForDisplay(role: AdminRole): string {
  return role === 'SUPERADMIN' ? 'a Super Admin' : 'an Admin';
}

/**
 * Generate invitation link for admin signup
 * @param token - Invitation token
 * @returns Full URL for admin signup with token
 */
export function generateInvitationLink(token: string): string {
  const baseUrl = process.env.NEXT_PUBLIC_APP_URL || 'http://localhost:3000';
  return `${baseUrl}/admin/signup?token=${token}`;
}