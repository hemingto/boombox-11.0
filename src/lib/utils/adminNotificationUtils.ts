/**
 * @fileoverview Utility functions for admin notifications
 * @source boombox-10.0/src/app/api/admin/notify-no-driver/route.ts (extracted inline functions)
 * @refactor Extracted utility functions for urgency calculation, address formatting, and template variables
 */

import { prisma } from '@/lib/database/prismaClient';

/**
 * Get admin emails from database for notifications
 * Fetches all ADMIN and SUPERADMIN role emails from the database
 */
export async function getAdminEmails(): Promise<string[]> {
  try {
    const admins = await prisma.admin.findMany({
      where: {
        role: {
          in: ['ADMIN', 'SUPERADMIN']
        }
      },
      select: {
        email: true
      }
    });

    const adminEmails = admins.map(admin => admin.email);
    
    // Fallback to environment variables if no admins found in database
    if (adminEmails.length === 0) {
      console.warn('No admin emails found in database, falling back to environment variables');
      return process.env.ADMIN_EMAILS?.split(',') || [
        'chemington@boomboxstorage.com',
        'help@boomboxstorage.com'
      ];
    }

    return adminEmails;
  } catch (error) {
    console.error('Error fetching admin emails from database:', error);
    // Fallback to environment variables on error
    return process.env.ADMIN_EMAILS?.split(',') || [
      'chemington@boomboxstorage.com',
      'help@boomboxstorage.com'
    ];
  }
}

/**
 * Determine urgency level based on delivery date
 */
export function getUrgencyLevel(deliveryDate: Date): 'critical' | 'high' | 'medium' {
  const now = new Date();
  const hoursUntilDelivery = (deliveryDate.getTime() - now.getTime()) / (1000 * 60 * 60);
  
  if (hoursUntilDelivery <= 4) return 'critical';
  if (hoursUntilDelivery <= 12) return 'high';
  return 'medium';
}

/**
 * Get urgency-specific styling colors
 */
export function getUrgencyColors(urgencyLevel: string) {
  const urgencyColors = {
    critical: { bg: '#fee2e2', border: '#dc2626', text: '#7f1d1d' },
    high: { bg: '#fef3c7', border: '#d97706', text: '#92400e' },
    medium: { bg: '#ede9fe', border: '#7c3aed', text: '#553c9a' },
  };

  return urgencyColors[urgencyLevel as keyof typeof urgencyColors] || urgencyColors.medium;
}

/**
 * Get urgency emoji based on level
 */
export function getUrgencyEmoji(urgencyLevel: string): string {
  const emojiMap = {
    critical: 'üö®',
    high: '‚ö†Ô∏è',
    medium: 'üìã'
  };
  
  return emojiMap[urgencyLevel as keyof typeof emojiMap] || 'üìã';
}

/**
 * Helper to get short address for display
 */
export function getShortAddress(fullAddress: string): string {
  const parts = fullAddress.split(',');
  const streetAddress = parts[0]?.trim() || '';
  
  if (streetAddress.length > 25) {
    return streetAddress.substring(0, 22) + '...';
  }
  
  return streetAddress;
}

/**
 * Generate orders section HTML for email template
 */
export function generateOrdersSection(routeDetails: any): string {
  if (!routeDetails?.orders || routeDetails.orders.length === 0) {
    return '';
  }

  const ordersTable = routeDetails.orders.map((order: any) => `
    <tr style="border-bottom: 1px solid #f3f4f6;">
      <td style="padding: 8px 4px; font-size: 14px;">#${order.id}</td>
      <td style="padding: 8px 4px; font-size: 14px;">${order.contactName}</td>
      <td style="padding: 8px 4px; font-size: 14px;">${getShortAddress(order.deliveryAddress)}</td>
      <td style="padding: 8px 4px; font-size: 14px; text-align: right;">$${order.totalPrice.toFixed(0)}</td>
    </tr>
  `).join('');

  return `
    <div style="margin: 20px 0;">
      <h4 style="color: #374151; margin: 0 0 12px 0; font-size: 16px;">üì¶ Order Details</h4>
      <div style="overflow-x: auto;">
        <table style="width: 100%; border-collapse: collapse; font-size: 14px; background: #f9fafb; border-radius: 6px;">
          <thead>
            <tr style="background: #f3f4f6;">
              <th style="padding: 8px 4px; text-align: left; font-weight: 600; color: #374151;">Order</th>
              <th style="padding: 8px 4px; text-align: left; font-weight: 600; color: #374151;">Customer</th>
              <th style="padding: 8px 4px; text-align: left; font-weight: 600; color: #374151;">Address</th>
              <th style="padding: 8px 4px; text-align: right; font-weight: 600; color: #374151;">Value</th>
            </tr>
          </thead>
          <tbody>
            ${ordersTable}
          </tbody>
        </table>
      </div>
    </div>
  `;
}

/**
 * Generate source context message based on notification source
 */
export function generateSourceContext(source?: string): string {
  if (source === 'expired_offer_cleanup') {
    return '<p style="color: #dc2626; font-weight: 600; margin: 8px 0;">‚è∞ Previous driver offer expired</p>';
  }
  return '';
}

/**
 * Generate source text for email subject
 */
export function generateSourceText(source?: string): string {
  return source === 'expired_offer_cleanup' ? ' (Expired Offer)' : '';
}

/**
 * Generate additional info section for email footer
 */
export function generateAdditionalInfoSection(additionalInfo?: string): string {
  if (!additionalInfo) {
    return '';
  }
  
  return `
    <p style="margin: 8px 0 0 0; color: #9ca3af; font-size: 11px; text-align: center;">
      Additional: ${additionalInfo}
    </p>
  `;
}

/**
 * Calculate estimated payout based on number of stops
 */
export function calculateEstimatedPayout(totalStops: number): string {
  return totalStops ? `$${(totalStops * 15).toFixed(0)}` : 'TBD';
}