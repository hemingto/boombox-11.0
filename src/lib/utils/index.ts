/**
 * @fileoverview Central utility exports for boombox-11.0
 * Organized by domain and functionality for clean imports
 */

// Core utility functions
export * from './cn';
export * from './phoneUtils';
export * from './currencyUtils';
export * from './navigationUtils';
export * from './emailUtils';

// Notification utilities
export * from './notificationUtils';
export * from './validationUtils';
export * from './formatUtils';
export * from './statusUtils';

// Blog utilities
export * from './blogUtils';

// Explicit exports to resolve conflicts
export {
  calculateMonthlyStorageCost,
  isInServiceArea,
  getServiceAreaForZip,
} from './businessUtils';

export {
  formatDateForInput,
  formatDateForDisplay,
  formatDateCompact,
  formatDateTime,
  formatTime,
  formatTime24Hour,
  formatTimestamp,
  parseAppointmentTime,
  calculateDeliveryWindow,
  getUnitSpecificStartTime,
  formatVerboseDate,
} from './dateUtils';

export {
  generateDriverToken,
  generateCustomerMoverChangeToken,
  notifyDriverReassignment,
  notifyCustomerMoverChange,
  recordDriverCancellation,
  recordMoverCancellation,
  type DriverCancellationData,
  type MoverCancellationData,
  type AvailableDriver,
} from './cancellationUtils';

// Domain-specific utilities (avoiding conflicts)
// Note: appointmentUtils contains parseLoadingHelpPrice which conflicts with pricingUtils
// Using pricingUtils version for parseLoadingHelpPrice
export {
  formatAppointmentTime,
  formatAppointmentDateForSms,
  formatTimeMinusOneHour,
  validateAppointmentDateTime,
  calculateAppointmentChanges,
  calculateFinalUnitCount,
  getUnitNumbersToRemove,
  getDayOfWeekForAvailability,
  validateMovingPartnerAvailability,
  generateDriverReconfirmToken,
  generateDriverWebViewUrl,
  formatDriverInfo,
  verifyTrackingToken,
  // parseLoadingHelpPrice - using pricingUtils version
} from './appointmentUtils';

// Explicit exports from availabilityUtils to avoid conflicts
export {
  DEFAULT_BUSINESS_HOURS,
  calculateDriverRequirements,
  formatTimeLocal,
  generateBusinessHourSlots,
  hasTimeConflict,
  isResourceAvailableInSlot,
  createTimeConflict,
  checkOnfleetTaskConflicts,
  determineAvailabilityLevel,
  getDistinctDaysOfWeekInMonth,
  groupBy,
  isDateInPast,
  generateCacheKey,
  getMinutesDifference,
  isValidTimeSlot,
  timeStringToMinutes,
  // getDayOfWeekString - conflicts with dateFormattingUtils, using that version
} from './availabilityUtils';

export * from './moverChangeUtils';

// Explicit exports from movingPartnerUtils to avoid conflicts
export {
  createDefaultMoverAvailability,
  parseTimeToMinutes,
  checkMoverExists,
  createMover,
  findAvailableMovingPartners,
  uploadFileToCloudinary,
  deleteOldCloudinaryFile,
  getMovingPartnerJobs,
  assignMovingPartnerDriver,
  // getDayOfWeekString - conflicts with other modules, already exported from dateFormattingUtils
} from './movingPartnerUtils';

// Messaging and communication utilities
// Note: twilioUtils contains server-side code (headers) - import directly in API routes only
export * from './messageClassificationUtils';
export * from './inboundMessageUtils';
// Note: dateFormattingUtils contains formatAppointmentTime which conflicts with appointmentUtils
// Using appointmentUtils version for appointment-specific formatting
// export * from './dateFormattingUtils';
// Explicit export of non-conflicting functions from dateFormattingUtils
export { formatAppointmentDate } from './dateFormattingUtils';

// Driver utilities (explicit exports to avoid conflicts)
export {
  getDriverTeamIds,
  createDefaultDriverAvailability,
  formatPhoneForOnfleet,
  mapVehicleTypeToOnfleet,
  buildOnfleetWorkerPayload,
  validateDriverUniqueness,
  findDriverInvitation,
  validateInvitationStatus,
  DEFAULT_DRIVER_SERVICES,
  DEFAULT_MOVING_PARTNER_VEHICLE_TYPE,
  DEFAULT_DRIVER_CAPACITY,
  DRIVER_STATUSES,
  type DriverRegistrationData,
  type OnfleetWorkerPayload,
  type DriverApprovalResult,
} from './driverUtils';

// Pricing utilities (explicit exports to avoid conflicts)
export {
  getBoomboxPriceByZipCode,
  formatStorageUnitPrice,
  formatInsurancePrice,
  parseLoadingHelpPrice, // Using pricingUtils version as canonical
} from './pricingUtils';

// Storage utilities (explicit exports)
export {
  getStorageUnitText,
} from './storageUtils';

// Sorting and pagination utilities
export * from './sortingUtils';

// Tracking utilities
export * from './trackingStatusUtils';
