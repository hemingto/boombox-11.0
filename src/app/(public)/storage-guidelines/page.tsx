/**
 * @fileoverview Storage guidelines page - what can/cannot be stored
 * @source boombox-10.0/src/app/storage-guidelines/page.tsx
 * @refactor Migrated to (public) route group with SEO metadata
 */

'use client';

import type { Metadata } from 'next';
import {
  StorageGuidelinesHero,
  StorageGuidelinesList,
} from '@/components/features/storage-guidelines';
import { FaqSection, HelpCenterSection } from '@/components/features/landing';

// Note: Page is client component
// SEO metadata should be added via generateMetadata when converting to server component

export default function StorageGuidelines() {
  return (
    <>
      <StorageGuidelinesHero />
      <StorageGuidelinesList />
      <FaqSection />
      <HelpCenterSection />
    </>
  );
}

