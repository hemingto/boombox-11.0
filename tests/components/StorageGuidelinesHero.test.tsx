/**
 * @fileoverview Tests for StorageGuidelinesHero component
 * Following boombox-11.0 testing standards (97â†’0 failure learnings)
 */
import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { testAccessibility } from '../utils/accessibility';
import { StorageGuidelinesHero } from '@/components/features/storage-guidelines/StorageGuidelinesHero';

expect.extend(toHaveNoViolations);

describe('StorageGuidelinesHero', () => {
  // REQUIRED: Test cleanup
  beforeEach(() => {
    jest.clearAllMocks();
  });

  // REQUIRED: Basic rendering
  describe('Rendering', () => {
    it('renders without crashing', () => {
      render(<StorageGuidelinesHero />);
      expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    });

    it('displays the correct heading text', () => {
      render(<StorageGuidelinesHero />);
      const heading = screen.getByRole('heading', { level: 1, name: /storage guidelines/i });
      expect(heading).toBeInTheDocument();
      expect(heading).toHaveTextContent('Storage Guidelines');
    });

    it('displays the subtitle text', () => {
      render(<StorageGuidelinesHero />);
      expect(screen.getByText('best practices')).toBeInTheDocument();
    });

    it('renders with correct container structure', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const outerDiv = container.firstChild;
      expect(outerDiv).toHaveClass('w-full', 'my-10', 'sm:my-20', 'lg:px-16', 'px-6');
    });
  });

  // MANDATORY: Accessibility testing
  describe('Accessibility', () => {
    it('has no accessibility violations', async () => {
      const renderResult = render(<StorageGuidelinesHero />);
      await testAccessibility(renderResult);
    });

    it('uses semantic HTML with proper heading hierarchy', () => {
      render(<StorageGuidelinesHero />);
      const heading = screen.getByRole('heading', { level: 1 });
      expect(heading.tagName).toBe('H1');
    });

    it('has accessible heading structure', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const headings = container.querySelectorAll('h1, h2, h3, h4, h5, h6');
      
      // Should have exactly one h1
      expect(headings).toHaveLength(1);
      expect(headings[0].tagName).toBe('H1');
    });

    it('paragraph text is readable', () => {
      render(<StorageGuidelinesHero />);
      const paragraph = screen.getByText('best practices');
      expect(paragraph.tagName).toBe('P');
    });
  });

  // Design System Integration
  describe('Design System Compliance', () => {
    it('applies responsive padding classes', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const outerDiv = container.firstChild as HTMLElement;
      
      // Test responsive padding
      expect(outerDiv).toHaveClass('lg:px-16');
      expect(outerDiv).toHaveClass('px-6');
    });

    it('applies responsive vertical spacing', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const outerDiv = container.firstChild as HTMLElement;
      
      // Test responsive margin
      expect(outerDiv).toHaveClass('my-10');
      expect(outerDiv).toHaveClass('sm:my-20');
    });

    it('maintains full width layout', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const outerDiv = container.firstChild as HTMLElement;
      expect(outerDiv).toHaveClass('w-full');
    });
  });

  // Component Structure
  describe('Component Structure', () => {
    it('renders nested div structure correctly', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const outerDiv = container.firstChild as HTMLElement;
      const innerDiv = outerDiv.firstChild as HTMLElement;
      
      expect(outerDiv.tagName).toBe('DIV');
      expect(innerDiv.tagName).toBe('DIV');
    });

    it('applies bottom margin to heading', () => {
      render(<StorageGuidelinesHero />);
      const heading = screen.getByRole('heading', { level: 1 });
      expect(heading).toHaveClass('mb-4');
    });

    it('contains both heading and paragraph elements', () => {
      const { container } = render(<StorageGuidelinesHero />);
      const heading = container.querySelector('h1');
      const paragraph = container.querySelector('p');
      
      expect(heading).toBeInTheDocument();
      expect(paragraph).toBeInTheDocument();
    });
  });

  // Content Verification
  describe('Content', () => {
    it('has the correct page title', () => {
      render(<StorageGuidelinesHero />);
      expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Storage Guidelines');
    });

    it('has the correct subtitle', () => {
      render(<StorageGuidelinesHero />);
      expect(screen.getByText(/best practices/i)).toBeInTheDocument();
    });
  });

  // Snapshot Testing
  describe('Snapshot', () => {
    it('matches snapshot', () => {
      const { container } = render(<StorageGuidelinesHero />);
      expect(container).toMatchSnapshot();
    });
  });
});

