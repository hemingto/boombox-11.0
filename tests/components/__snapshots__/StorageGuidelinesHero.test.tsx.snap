// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StorageGuidelinesHero Snapshot matches snapshot 1`] = `
<div>
  <div
    class="w-full my-10 sm:my-20 lg:px-16 px-6"
  >
    <div>
      <h1
        class="mb-4"
      >
        Storage Guidelines
      </h1>
      <p>
        best practices
      </p>
    </div>
  </div>
</div>
`;
